src/routes/auth/auth.controller.ts

```diff
@@ -32,12 +35,26 @@ export class AuthController {
   @HttpCode(200)
   @Post('verify')
   async verify(
+    @Res({ passthrough: true })
+    res: Response,
     @Body(new ValidationPipe(VerifyAuthMessageDtoSchema))
     verifyAuthMessageDto: VerifyAuthMessageDto,
-  ): Promise<{
-    accessToken: string;
-    tokenType: string;
-  }> {
-    return await this.authService.verify(verifyAuthMessageDto);
+  ): Promise<void> {
+    const { accessToken } =
+      await this.authService.getAccessToken(verifyAuthMessageDto);
+
+    const maxAge = verifyAuthMessageDto.message.expirationTime
+      ? getMillisecondsUntil(
+          new Date(verifyAuthMessageDto.message.expirationTime),
+        )
+      : undefined;
+
+    res.cookie(AuthController.ACCESS_TOKEN_COOKIE_NAME, accessToken, {
+      httpOnly: true,
```

Comment:
One thing that we should consider is the support of HTTP only cookies for non-browser clients (e.g.: mobile). In this case the cookie management won't be handled by the browser but by the HTTP client itself.

From what I know, if we return this in a cookie, this state won't be handled automatically on non-browser clients so additional work needs to be done there.

Reply:
I'll be sure to make a note of this when documenting the implementation.